rmw_connext_cpp/src/rmw_init.cpp

```diff
@@ -105,12 +110,27 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
     return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
   context->instance_id = options->instance_id;
   context->implementation_identifier = rti_connext_identifier;
-  context->impl = nullptr;
+  context->impl = new (std::nothrow) rmw_context_impl_t();
+  if (!context->impl) {
+    RMW_SET_ERROR_MSG("failed to allocate context impl");
+    return RMW_RET_BAD_ALLOC;
+  }
+  context->impl->is_shutdown = false;
   rmw_ret_t ret = rmw_init_options_copy(options, &context->options);
   if (RMW_RET_OK != ret) {
+    delete context->impl;
     return ret;
   }
-  return init();
+  ret = init();
+  if (RMW_RET_OK != ret) {
+    if (RMW_RET_OK != rmw_init_options_fini(&context->options)) {
```

Comment:
this seem unrelated to the PR.
IMO, it should be done in another PR, or in a separate commit.

Reply:
That is fair, yeah. I came across leaks and I couldn't help it.